UserDataProcessingTrigger/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { createTableService } from "azure-storage";
+import { getConfigOrThrow } from "../utils/config";
+import { deleteTableEntity, insertTableEntity } from "../utils/storage";
+
+// prepare table storage utils
+const config = getConfigOrThrow();
+const connectionString = config.FailedUserDataProcessingStorageConnection;
+const failedUserDataProcessingTable = config.FAILED_USER_DATA_PROCESSING_TABLE;
+
+const tableService = createTableService(connectionString);
+
+export const addFailedUserDataProcessing = insertTableEntity(
+  tableService,
+  failedUserDataProcessingTable
+);
+
+export const removeFailedUserDataProcessing = deleteTableEntity(
+  tableService,
+  failedUserDataProcessingTable
+);
+
+export type InsertTableEntityType = typeof addFailedUserDataProcessing;
+
+export type DeleteTableEntityType = typeof removeFailedUserDataProcessing;
+
+export const createFailedUserDataProcessingTableIfNotExists = (): void =>
+  tableService.createTableIfNotExists(failedUserDataProcessingTable, () => 0);
```

Comment:
In effect I've seen this done somewhere else without any failure check. Anyway without the table the insertEntity will return 404 and the process will fail later returning an error. Should we check and return this error before?

Reply:
I added a promisified version of createTableIfNotExists, I call when the function starts and if it returns an error I log it. I would not try to break the function because it would break the user data processing stages. As said, insertEntity will fail and log the error occurred if no table is found.